internal/api/mfa.go

```diff
@@ -263,6 +236,41 @@ func (a *API) EnrollFactor(w http.ResponseWriter, r *http.Request) error {
 	})
 }
 
+func (a *API) EnrollFactor(w http.ResponseWriter, r *http.Request) error {
+	ctx := r.Context()
+	user := getUser(ctx)
+	session := getSession(ctx)
+	config := a.config
+
+	if session == nil || user == nil {
+		return internalServerError("A valid session and a registered user are required to enroll a factor")
+	}
+	params := &EnrollFactorParams{}
+	if err := retrieveRequestParams(r, params); err != nil {
+		return err
+	}
+
+	switch params.FactorType {
+	case models.Phone:
+		if !config.MFA.Phone.EnrollEnabled {
+			return unprocessableEntityError(ErrorCodeMFAPhoneEnrollDisabled, "MFA enroll is disabled for Phone")
+		}
+		return a.enrollPhoneFactor(w, r, params)
+	case models.TOTP:
+		// Prior to the introduction of MFA.TOTP.EnrollEnabled,
+		// MFA.Enabled was used to configure whether TOTP was on. So
+		// both have to be set to false to regard the feature as
+		// disabled.
+		if !config.MFA.Enabled && !config.MFA.TOTP.EnrollEnabled {
```

Comment:
would be nice if we can get `MFA.TOTP.EnrollEnabled` to reference `GOTRUE_MFA_TOTP_ENROLL_ENABLED` but fallback to `GOTRUE_MFA_ENROLL_ENABLED` if the former is undefined

that way, we can just use 1 condition here, instead of having to set both to false

Reply:
Makes sense, would prefer to do it in a FLUP PR though. 

This entire PR is a copy paste - original change [here](https://github.com/supabase/auth/pulls?q=sort%3Aupdated-desc+is%3Apr+is%3Aclosed0